lib/ansible/utils/path.py

```diff
@@ -127,7 +127,7 @@ def cleanup_tmp_file(path, warn=False):
                 elif os.path.isfile(path):
                     os.unlink(path)
             except Exception as e:
-                if warn:
+                if warn and getattr(e, 'errno', None) != ENOENT:
```

Comment:
```suggestion
            except OSError as e:
                if warn and e.errno != ENOENT:
```

Reply:
I cannot remember exactly why this change was made. I can only surmise that I saw some form of error. More research would be needed. You can also just try removing it to see what happens.